neqo-http3/src/connection_client.rs

```diff
@@ -515,6 +516,46 @@ impl Http3Client {
         output
     }
 
+    /// The function establishes a classic HTTP CONNECT tunnel using `target`
+    /// and `headers`. Data can be send into the tunnel via
+    /// [`Http3Client::send_data`] and received from the tunnel via
+    /// [`Http3Client::read_data`]. The tunnel can be closed via
+    /// [`Http3Client::stream_close_send`].
+    ///
+    /// # Errors
+    ///
+    /// If a new stream cannot be created an error will be return.
+    pub fn classic_connect<'x, 't: 'x, T>(
```

Comment:
Thinking about the name here...  I think that we want:

* `fn connect()`
* `fn connect_webtransport()`
* `fn connect_udp()`
* and maybe `connect_tcp()` (though I remain skeptical there)

I know that `connection.connect()` has some awkward implications for the connection itself, but at this level, I think we are OK to use the HTTP connect.

Reply:
Done in https://github.com/mozilla/neqo/pull/2886/commits/694c2b28dfbfa1107124a90133686131df3ae46b.